projects/igniteui-angular/src/lib/simple-combo/simple-combo.component.ts

```diff
@@ -402,10 +394,16 @@ export class IgxSimpleComboComponent extends IgxComboBaseDirective implements Co
             owner: this,
             cancel: false
         };
-        this.selectionChanging.emit(args);
+        // additional checks when selecting and clearing an item with valueKey=undefined
+        // as the event should emit when args.newSelection differs from args.oldSelection
+        // however in this case both args.newSelection and args.oldSelection are 'undefined'
+        if (args.newSelection !== args.oldSelection
+            || args.newSelection === undefined && newSelection?.size > 0
+            || newSelection?.size === 0 && oldSelectionAsArray.length > 0) {
+            this.selectionChanging.emit(args);
+        }
```

Comment:
do not proceed with selection update when `newSelection` is same as `oldSelection`.